### PR TITLE
Add past contributions table to contributions

### PR DESCRIPTION
**Pull request by kdv24**

### CODE REVIEW

File: app/src/components/PreviousDonations/PreviousDonationsTable.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      state.governments.currentGovernmentId,
+    campaignId: state.campaigns.currentCampaignId,
+    govId: state.governments.currentGovernmentId || 1,
+    userId: isLoggedIn(state) ? state.auth.me.id : null,
+  }),
+  dispatch => {
+    return {
+      getFilterOptions,
+      updateFilter: filterOptions => dispatch(updateFilter(filterOptions)),
+      pastContributions: matchId => dispatch(pastContributions(matchId)),
```
Comment (punkael) [inline app/src/components/PreviousDonations/PreviousDonationsTable.js]: getContributionsByMatchId is the function you want to dispatch

File: app/src/components/PreviousDonations/PreviousDonationsTable.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PageHoc from '../PageHoc/PageHoc';
+import Table from '../Table';
+import Button from '../Button/Button';
+import {
+  getContributionsList,
+  getContributionsTotal,
```
Comment (punkael) [inline app/src/components/PreviousDonations/PreviousDonationsTable.js]: Likely don't need filters

File: app/src/components/PreviousDonations/PreviousDonationsTable.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    ? { icon: 'none', name, buttonType, onClick, isFreeAction }
+    : { icon: 'none', name, buttonType, onClick };
+
+const columns = isGovAdmin => [
+  {
+    field: 'name',
+    title: 'Name',
+  },
+  {
+    ...(isGovAdmin
```
Comment (punkael) [inline app/src/components/PreviousDonations/PreviousDonationsTable.js]: Only need govAdmin fields

File: app/src/components/PreviousDonations/PreviousDonationsTable.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+          totalRows={total}
+        />
+      </PageHoc>
+    );
+  }
+}
+
+export default connect(
+  state => ({
+    filterOptions: getFilterOptions(state),
```
Comment (punkael) [inline app/src/components/PreviousDonations/PreviousDonationsTable.js]: Remove all props not required

File: app/src/components/Forms/ContributionReady/index.js [hunk 5]
```diff
@@ -167,6 +171,21 @@ class ContributionReadyForm extends React.Component {
                   contributionId={values.id}
                   showOccupationLetter={visibleIf.showOccupationLetter}
                 />
+                {console.log({ matchId })}
+                {/* {isGovAdmin && matchId ? ( // ----KELLY - how call match id? */}
+                <div>
+                  <h2>Previous Donations</h2>
+                  <PreviousDonationsTable
+                    pastContributions={pastContributions}
```
Comment (punkael) [inline app/src/components/Forms/ContributionReady/index.js]: Looks like pastContributions is already called if there is a matchId in contribution duct. So no need to add it, pass it, or call it.

File: app/src/components/Forms/ContributionReady/index.js [hunk 6]
```diff
@@ -191,5 +210,6 @@ export default connect(
       dispatch(flashMessage(message, options)),
     updateContribution: data => dispatch(updateContribution(data)),
     getContributionById: id => dispatch(getContributionById(id)),
+    pastContributions: matchId => dispatch(pastContributions[matchId]),
```
Comment (punkael) [inline app/src/components/Forms/ContributionReady/index.js]: Remove not needed ^^

File: app/src/components/PreviousDonations/PreviousDonationsTable.js (hunk 2 above)
Comment (kdv24) [reply app/src/components/PreviousDonations/PreviousDonationsTable.js]: yeah, those have come out.  thx.

File: app/src/components/PreviousDonations/PreviousDonationsTable.js (hunk 1 above)
Comment (punkael) [inline app/src/components/PreviousDonations/PreviousDonationsTable.js]: Correction: The correct function to dispatch would be getContributionsByMatchId, but you don't need to dispatch it, it is already dispatched automatically when the contribution is loaded.

File: app/src/components/PreviousDonations/PreviousDonationsTable.js (hunk 3 above)
Comment (kdv24) [reply app/src/components/PreviousDonations/PreviousDonationsTable.js]: 👍

File: app/src/components/Forms/ContributionReady/index.js (hunk 5 above)
Comment (kdv24) [reply app/src/components/Forms/ContributionReady/index.js]: ok.

File: app/src/components/Forms/ContributionReady/index.js [hunk 7]
```diff
@@ -167,6 +169,18 @@ class ContributionReadyForm extends React.Component {
                   contributionId={values.id}
                   showOccupationLetter={visibleIf.showOccupationLetter}
                 />
+                {console.log({ matchId })}
```
Comment (punkael) [inline app/src/components/Forms/ContributionReady/index.js]: Try {console.log( currentContribution.matchId )}

Comment (kdv24) [reply app/src/components/Forms/ContributionReady/index.js]: Just now seeing this one, so I'd already gotten it.  Thanks though!

**Comment by kdv24** [comment]
closes #619

File: app/src/components/PreviousDonations/PreviousDonationsTable.js [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+          : history.push(`/contributions/${rowData.id}`);
+      }),
+    ];
+
+    const components = {
+      // eslint-disable-next-line react/display-name
+      Action: props => (
+        <Button
+          onClick={event => props.action.onClick(event, props.data)}
+          buttonType={props.action.buttonType}
```
Comment (punkael) [inline app/src/components/PreviousDonations/PreviousDonationsTable.js:65]: This is great. Should be able to do something like this on the button properties, if disabling is a desired effect.
disabled={!!(props.data.id === currentId)}

File: app/src/components/PreviousDonations/PreviousDonationsTable.js [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+            components={components}
+            data={pastContributions}
+          />
+        )}
+      </PageHoc>
+    );
+  }
+}
+
+export default connect()(PreviousDonationsTable);
```
Comment (punkael) [inline app/src/components/PreviousDonations/PreviousDonationsTable.js]: Does the conect do anything here? Might be able to remove it.

File: app/src/components/PreviousDonations/PreviousDonationsTable.js [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+          day: '2-digit',
+        })
+        .split(', ')[0],
+  },
+];
+
+class PreviousDonationsTable extends React.Component {
+  render() {
+    const {
+      isListLoading,
```
Comment (punkael) [inline app/src/components/PreviousDonations/PreviousDonationsTable.js]: isListLoading may be undefined, could be removed.

File: app/src/components/PreviousDonations/PreviousDonationsTable.js [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+      history,
+      pastContributions,
+      matchId,
+      currentId,
+    } = this.props;
+
+    const isLoading = isListLoading && !Array.isArray(pastContributions);
```
Comment (punkael) [inline app/src/components/PreviousDonations/PreviousDonationsTable.js]: If undefined above^ remove isListLoading here as well.

File: app/src/components/PreviousDonations/PreviousDonationsTable.js (hunk 9 above)
Comment (kdv24) [reply app/src/components/PreviousDonations/PreviousDonationsTable.js]: yep.

File: app/src/components/PreviousDonations/PreviousDonationsTable.js (hunk 10 above)
Comment (kdv24) [reply app/src/components/PreviousDonations/PreviousDonationsTable.js]: you're right

**Pull request merged**